sunpy/map/mapbase.py

```diff
@@ -1776,7 +1779,7 @@ def rotate(self, angle: u.deg = None, rmatrix=None, order=3, scale=1.0,
         return new_map
 
     @u.quantity_input
-    def submap(self, bottom_left, *, top_right=None, width: (u.deg, u.pix) = None, height: (u.deg, u.pix) = None):
+    def submap(self, bottom_left, *, top_right=None, width: _width_height_type = None, height: _width_height_type = None):
```

Comment:
I feel like this change has had a negative effect on our documentation.

Before:
![2023-10-09-144642_grim](https://github.com/sunpy/sunpy/assets/1391051/6c706511-4cd0-4c37-9522-ba60c779efaa)

After:
![2023-10-09-144739_grim](https://github.com/sunpy/sunpy/assets/1391051/6f0cec52-d332-4bff-be00-cdf578b6ba0d)

Now I am not saying either is great but it's worse now.

Perhaps we should look into using this: https://pypi.org/project/sphinx-autodoc-typehints/